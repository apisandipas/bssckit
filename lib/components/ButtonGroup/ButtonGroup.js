"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _theme = _interopRequireDefault(require("../../theme"));

var _base = require("../Utilities/base");

var _themeFunctions = require("../../utils/themeFunctions");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n  position: relative;\n  display: inline-flex;\n  vertical-align: middle;\n  & > button {\n    line-height: 1.5;\n    position: relative;\n    flex: 0 1 auto;\n    &:hover {\n      z-index: 1;\n    }\n    ", ";\n  }\n  ", ";\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n    & > button {\n      &:not(:last-child) {\n        border-top-right-radius: 0;\n        border-bottom-right-radius: 0;\n      }\n      &:not(:first-child) {\n        border-top-left-radius: 0;\n        border-bottom-left-radius: 0;\n      }\n    }\n  "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n      flex-direction: column;\n      align-items: flex-start;\n      justify-content: center;\n      & > button {\n        width: 100%;\n        &:not(:last-child) {\n          border-bottom-right-radius: 0;\n          border-bottom-left-radius: 0;\n        }\n        &:not(:first-child) {\n          border-top-left-radius: 0;\n          border-top-right-radius: 0;\n        }\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n    padding: ", ";\n    font-size: ", ";\n    line-height: 1.5;\n  "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n      padding: ", ";\n      font-size: ", ";\n      line-height: 1.5;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n      padding: ", ";\n      font-size: ", ";\n      line-height: 1.5;\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var size = function size(props) {
  if (props.lg) {
    return (0, _styledComponents.css)(_templateObject(), (0, _themeFunctions.getPadding)(props, 'buttonGroup', 'lg'), (0, _themeFunctions.getFontSize)(props, 'buttonGroup', 'lg'));
  } else if (props.sm) {
    return (0, _styledComponents.css)(_templateObject2(), (0, _themeFunctions.getPadding)(props, 'buttonGroup', 'sm'), (0, _themeFunctions.getFontSize)(props, 'buttonGroup', 'sm'));
  }

  return (0, _styledComponents.css)(_templateObject3(), (0, _themeFunctions.getPadding)(props, 'buttonGroup', 'default'), (0, _themeFunctions.getFontSize)(props, 'buttonGroup', 'default'));
};

var vertical = function vertical(props) {
  if (props.vertical) {
    return (0, _styledComponents.css)(_templateObject4());
  }

  return (0, _styledComponents.css)(_templateObject5());
};

var ButtonGroup = (0, _styledComponents["default"])(_base.Div)(_templateObject6(), function (props) {
  return size(props);
}, function (props) {
  return vertical(props);
});
ButtonGroup.defaultProps = {
  theme: _theme["default"]
};
var _default = ButtonGroup;
exports["default"] = _default;