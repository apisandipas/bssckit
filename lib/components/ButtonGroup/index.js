"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ButtonGroup", {
  enumerable: true,
  get: function get() {
    return _ButtonGroup["default"];
  }
});
Object.defineProperty(exports, "ButtonToolbar", {
  enumerable: true,
  get: function get() {
    return _ButtonToolbar["default"];
  }
});

var _ButtonGroup = _interopRequireDefault(require("./ButtonGroup"));

var _ButtonToolbar = _interopRequireDefault(require("./ButtonToolbar"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }