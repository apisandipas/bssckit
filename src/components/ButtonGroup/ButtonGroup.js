import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { theme } from 'styled-tools';
import defaultTheme from '../../theme';
import { Div } from '../base';

const getSize = props => {
  if (props.lg) {
    return css`
      padding: ${theme('buttonGroup.padding.lg')};
      font-size: ${theme('buttonGroup.fontSize.lg')};
      line-height: 1.5;
    `;
  } else if (props.sm) {
    return css`
      padding: ${theme('buttonGroup.padding.sm')};
      font-size: ${theme('buttonGroup.fontSize.sm')};
      line-height: 1.5;
    `;
  }

  return css`
    padding: ${theme('buttonGroup.padding.default')};
    font-size: ${theme('buttonGroup.fontSize.default')};
    line-height: 1.5;
  `;
};

const getVertical = props => {
  if (props.vertical) {
    return css`
      flex-direction: column;
      align-items: flex-start;
      justify-content: center;
      & > button {
        width: 100%;
        &:not(:last-child) {
          border-bottom-right-radius: 0;
          border-bottom-left-radius: 0;
        }
        &:not(:first-child) {
          border-top-left-radius: 0;
          border-top-right-radius: 0;
        }
      }
    `;
  }

  return css`
    & > button {
      &:not(:last-child) {
        border-top-right-radius: 0;
        border-bottom-right-radius: 0;
      }
      &:not(:first-child) {
        border-top-left-radius: 0;
        border-bottom-left-radius: 0;
      }
    }
  `;
};

const ButtonGroup = styled(Div)`
  position: relative;
  display: inline-flex;
  vertical-align: middle;
  & > button {
    line-height: 1.5;
    position: relative;
    flex: 0 1 auto;
    &:hover {
      z-index: 1;
    }
    ${props => getSize(props)};
  }
  ${props => getVertical(props)};
`;

ButtonGroup.propTypes = {
  theme: PropTypes.object.isRequired,
};

ButtonGroup.defaultProps = {
  theme: defaultTheme,
};

export default ButtonGroup;
