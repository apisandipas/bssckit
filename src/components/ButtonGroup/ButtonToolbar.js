import styled from 'styled-components';
import defaultTheme from '../../theme';
import { Div } from '../Utilities/base';

const ButtonToolbar = styled(Div)`
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  width: auto;
`;

ButtonToolbar.defaultProps = {
  theme: defaultTheme,
};

export default ButtonToolbar;
