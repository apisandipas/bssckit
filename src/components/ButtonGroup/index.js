export { default as ButtonGroup } from './ButtonGroup';
export { default as ButtonToolbar } from './ButtonToolbar';
