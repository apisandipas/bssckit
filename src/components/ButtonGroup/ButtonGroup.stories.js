import React from 'react';
import { Button } from '../Button';
import ButtonGroup from './ButtonGroup';
import ButtonToolbar from './ButtonToolbar';

export default {
  title: 'Button Group',
};

export const simple = () => (
  <>
    <ButtonGroup>
      <Button>Left</Button>
      <Button secondary>Middle</Button>
      <Button>Right</Button>
    </ButtonGroup>
  </>
);

export const sized = () => (
  <>
    <div>
      <ButtonGroup lg mb2>
        <Button secondary>Left</Button>
        <Button secondary>Middle</Button>
        <Button secondary>Right</Button>
      </ButtonGroup>
    </div>
    <div>
      <ButtonGroup mb2>
        <Button secondary>Left</Button>
        <Button secondary>Middle</Button>
        <Button secondary>Right</Button>
      </ButtonGroup>
    </div>
    <div>
      <ButtonGroup sm dBlock mb2>
        <Button secondary>Left</Button>
        <Button secondary>Middle</Button>
        <Button secondary>Right</Button>
      </ButtonGroup>
    </div>
  </>
);

export const vertical = () => (
  <ButtonGroup vertical>
    <Button secondary>Button</Button>
    <Button secondary>Button</Button>
    <Button secondary>Button</Button>
    <Button secondary>Button</Button>
    <Button secondary>Button</Button>
    <Button secondary>Button</Button>
  </ButtonGroup>
);

export const toolbar = () => (
  <ButtonToolbar>
    <ButtonGroup mr2>
      <Button>1</Button>
      <Button>2</Button>
      <Button>3</Button>
      <Button>4</Button>
    </ButtonGroup>
    <ButtonGroup mr2>
      <Button>5</Button>
      <Button>6</Button>
      <Button>7</Button>
    </ButtonGroup>
    <ButtonGroup>
      <Button>8</Button>
    </ButtonGroup>
  </ButtonToolbar>
);
