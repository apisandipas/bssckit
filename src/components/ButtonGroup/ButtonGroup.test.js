import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';

import ButtonGroup from './ButtonGroup';

describe('ButtonGroup Component', () => {
  it('matches ButtonGroup', () => {
    const { container } = render(<ButtonGroup />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('position', 'relative');
    expect(container.firstChild).toHaveStyleRule('display', 'inline-flex');
    expect(container.firstChild).toHaveStyleRule('vertical-align', 'middle');
    expect(container.firstChild).toHaveStyleRule('line-height', '1.5', {
      modifier: '& > button',
    });
    expect(container.firstChild).toHaveStyleRule('position', 'relative', {
      modifier: '& > button',
    });
    expect(container.firstChild).toHaveStyleRule('flex', '0 1 auto', {
      modifier: '& > button',
    });
  });

  it('matches ButtonGroup lg', () => {
    const { container } = render(<ButtonGroup lg />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('padding', '0.5rem 1rem', {
      modifier: '& > button',
    });
    expect(container.firstChild).toHaveStyleRule('font-size', '1.25rem', {
      modifier: '& > button',
    });
    expect(container.firstChild).toHaveStyleRule('line-height', '1.5', {
      modifier: '& > button',
    });
  });

  it('matches ButtonGroup sm', () => {
    const { container } = render(<ButtonGroup sm />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('padding', '0.25rem 0.5rem', {
      modifier: '& > button',
    });
    expect(container.firstChild).toHaveStyleRule('font-size', '0.875rem', {
      modifier: '& > button',
    });
    expect(container.firstChild).toHaveStyleRule('line-height', '1.5', {
      modifier: '& > button',
    });
  });

  it('matches ButtonGroup vertical', () => {
    const { container } = render(<ButtonGroup vertical />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('flex-direction', 'column');
    expect(container.firstChild).toHaveStyleRule('align-items', 'flex-start');
    expect(container.firstChild).toHaveStyleRule('justify-content', 'center');
    expect(container.firstChild).toHaveStyleRule('width', '100%', {
      modifier: '& > button',
    });
  });
});
