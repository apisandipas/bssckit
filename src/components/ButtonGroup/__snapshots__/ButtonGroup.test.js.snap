// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup Component matches ButtonGroup 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
}

.c0 > button {
  line-height: 1.5;
  position: relative;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
}

.c0 > button:hover {
  z-index: 1;
}

.c0 > button:not(:last-child) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.c0 > button:not(:first-child) {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
/>
`;

exports[`ButtonGroup Component matches ButtonGroup lg 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
}

.c0 > button {
  line-height: 1.5;
  position: relative;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  padding: 0.5rem 1rem;
  font-size: 1.25rem;
  line-height: 1.5;
}

.c0 > button:hover {
  z-index: 1;
}

.c0 > button:not(:last-child) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.c0 > button:not(:first-child) {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
/>
`;

exports[`ButtonGroup Component matches ButtonGroup sm 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
}

.c0 > button {
  line-height: 1.5;
  position: relative;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  padding: 0.25rem 0.5rem;
  font-size: 0.875rem;
  line-height: 1.5;
}

.c0 > button:hover {
  z-index: 1;
}

.c0 > button:not(:last-child) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.c0 > button:not(:first-child) {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
/>
`;

exports[`ButtonGroup Component matches ButtonGroup vertical 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: middle;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > button {
  line-height: 1.5;
  position: relative;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
}

.c0 > button:hover {
  z-index: 1;
}

.c0 > button {
  width: 100%;
}

.c0 > button:not(:last-child) {
  border-bottom-right-radius: 0;
  border-bottom-left-radius: 0;
}

.c0 > button:not(:first-child) {
  border-top-left-radius: 0;
  border-top-right-radius: 0;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<div
  class="sc-fznyAO c0"
/>
`;
